test/e2e/framework/job/wait.go

```diff
@@ -133,3 +149,23 @@ func WaitForAllJobPodsGone(ctx context.Context, c clientset.Interface, ns, jobNa
 		return len(pods.Items) == 0, nil
 	})
 }
+
+// WaitForJobCondition waits for a job to be matched to the given condition.
+// The condition callback may use gomega.StopTrying to abort early.
+func WaitForJobCondition(ctx context.Context, c clientset.Interface, ns, jobName, conditionDesc string, timeout time.Duration, condition jobCondition) error {
+	return framework.Gomega().
+		Eventually(ctx, framework.RetryNotFound(framework.GetObject(c.BatchV1().Jobs(ns).Get, jobName, metav1.GetOptions{}))).
+		WithTimeout(timeout).
+		Should(framework.MakeMatcher(func(job *batchv1.Job) (func() string, error) {
+			done, err := condition(job)
+			if err != nil {
+				return nil, err
+			}
+			if done {
+				return nil, nil
+			}
+			return func() string {
+				return fmt.Sprintf("expected job to be %s, got instead:\n%s", conditionDesc, format.Object(job, 1))
```

Comment:
what is the `1` for?

Reply:
```
/*
Pretty prints the passed in object at the passed in indentation level.

Object recurses into deeply nested objects emitting pretty-printed representations of their components.

Modify format.MaxDepth to control how deep the recursion is allowed to go
Set format.UseStringerRepresentation to true to return object.GoString() or object.String() when available instead of
recursing into the object.

Set PrintContextObjects to true to print the content of objects implementing context.Context
*/
func Object(object interface{}, indentation uint) string {
	indent := strings.Repeat(Indent, int(indentation))
	value := reflect.ValueOf(object)
	commonRepresentation := ""
	if err, ok := object.(error); ok && !isNilValue(value) { // isNilValue check needed here to avoid nil deref due to boxed nil
		commonRepresentation += "\n" + IndentString(err.Error(), indentation) + "\n" + indent
	}
	return fmt.Sprintf("%s<%s>: %s%s", indent, formatType(value), commonRepresentation, formatValue(value, indentation))
}
```

Reply:
`1` sounds useless then? we wouldn't see the conditions. But that's a nit.